subprojects/test-runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt

```diff
@@ -40,54 +45,70 @@ import kotlinx.coroutines.channels.ReceiveChannel
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.TestCoroutineScope
 import kotlinx.coroutines.test.runBlockingTest
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.io.TempDir
 import java.io.File
 import java.util.concurrent.TimeUnit
 
 @ExperimentalCoroutinesApi
 internal class RunnerIntegrationTest {
 
-    private val devices = Channel<Device>(Channel.UNLIMITED)
+    private val testCoroutineDispatcher = TestCoroutineDispatcher()
```

Comment:
I got so many problems when fixing this test. 
Some tests were hanging, some failed for unclear reasons.
Notices I made:
- Make sure that you everywhere use the same TestDispatcher
- Do not try to analyze code - use debugger